compiler/rustc_mir_dataflow/src/move_paths/builder.rs

```diff
@@ -112,6 +112,9 @@ impl<'b, 'a, 'tcx> Gatherer<'b, 'a, 'tcx> {
         let mut union_path = None;
 
         for (place_ref, elem) in data.rev_lookup.un_derefer.iter_projections(place.as_ref()) {
+            if elem.is_subtype() {
+                continue;
+            }
```

Comment:
Why?

Reply:
Otherwise we get `No move_errors should be allowed in MIR borrowck`, which shouldn't happen(?) with changes introduced in this PR.